### DISCUSSION TITLE
Build error: clamp

### DISCUSSION THREAD

**Discussion by vincent-lg** [Blightmud/Blightmud#278] [Q&A] [answered]
Hello,

In the last commit (f9888efc988e5d074b82c62b8226a5cb9a8c2164), when running `cargo build`, I hit an error:

```
   Compiling blightmud v3.1.0 (/home/pi/src/Blightmud)
error[E0658]: use of unstable library feature 'clamp'
  --> src/ui/screen.rs:74:29
   |
74 |         let height = height.clamp(1, 5);
   |                             ^^^^^
   |
   = note: see issue #44095 <https://github.com/rust-lang/rust/issues/44095> for more information

error[E0658]: use of unstable library feature 'clamp'
  --> src/ui/screen.rs:86:41
   |
86 |         self.status_lines.resize(height.clamp(1, 5) as usize, None);
   |                                         ^^^^^
   |
   = note: see issue #44095 <https://github.com/rust-lang/rust/issues/44095> for more information

error: aborting due to 2 previous errors
```

I'm not sure what it means, though I've tried to skim through the issue that was printed out.  Seems pretty technical to me.  Is there any fix?  Should I build in unstable?

The README file doesn't mention that, but perhaps another file does.

Thanks,

**Comment by LiquidityC** [comment, marked as answer]
We are running on the latest version of rust. Run `rustup update stable` or by other means update your rust installation and it should work fine.
